Docs: .NET Reference `after` function example updates (#509)
amplication/docs: docs/plugins/before-after.md

) => Promisable<ModuleMap>;

// .NET [version](https://github.com/amplication/amplication/blob/master/libs/util/code-gen-types/src/dotnet-plugins.types.ts#L22)
export type PluginAfterEvent<T extends EventParams, F> = (
  dsgContext: DsgContext,
  eventParams: T,
  files: FileMap<F>
) => Promisable<FileMap<F>>;
```

In the `before` and `after` functions, we have an access to the context and the event params.
The [context](docs\plugins\context.md) is used to gather common parts between events.
The event params manipulate the default behavior by passing different values.

## Accessing and Modifying Generated Files

In the `after` function, we also have access to the generated files. An example of using this parameter is when you want to restructure the generated files in a different folder structure.

:::info
In the `after` function for .NET plugins, we use a [`FileMap`](https://github.com/amplication/amplication/blob/master/libs/util/code-gen-types/src/files/file-map.ts). For Node.js plugins' `after` function we use a [`ModuleMap`](https://github.com/amplication/amplication/blob/master/libs/util/code-gen-types/src/code-gen-types.ts#L149).
:::

## Function Examples

> Most of the functions include examples. But, if you're looking for more comprehensive and real-world examples, you can explore the [code of the Amplication plugins](https://github.com/amplication/plugins/tree/master/plugins)directly. These plugins showcase various implementations and use cases for the before and after lifecycle functions.

Reviewer: ```suggestion
Most of the functions include examples. But, if you're looking for more comprehensive and real-world examples, you can explore the [code of the Amplication plugins](https://github.com/amplication/plugins/tree/master/plugins) directly. These plugins showcase various implementations and use cases for the before and after lifecycle functions.
```

  eventParams: T
) => Promisable<T>;

> // Node.js version

Reviewer: ```suggestion
// Node.js [version](https://github.com/amplication/amplication/blob/master/libs/util/code-gen-types/src/plugins.types.ts#L21)
```

> export type PluginAfterEventNodejs<T extends EventParams> = (

Reviewer: ```suggestion
export type PluginAfterEvent<T extends EventParams> = (
```

  dsgContext: DsgContext,
  eventParams: T,
  modules: ModuleMap
) => Promisable<ModuleMap>;

> // .NET version

Reviewer: ```suggestion
// .NET [version](https://github.com/amplication/amplication/blob/master/libs/util/code-gen-types/src/dotnet-plugins.types.ts#L22)
```

> export type PluginAfterEventDotNet<T extends EventParams, F> = (

Reviewer: ```suggestion
export type PluginAfterEvent<T extends EventParams, F> = (
```

  dsgContext: DsgContext,
  eventParams: T,
  files: FileMap<F>
) => Promisable<FileMap<F>>;
```

In the `before` and `after` functions, we have an access to the context and the event params.
The [context](docs\plugins\context.md) is used to gather common parts between events.
The event params manipulate the default behavior by passing different values.

In the `after` function, we also have access to the generated files. An example of using this parameter is when you want to restructure the generated files in a different folder structure.

:::info

> In the `after` function for .NET plugins, we use a [`FileMap`](https://github.com/amplication/amplication/blob/master/libs/util/code-gen-types/src/files/index.ts). For Node.js plugins' `after` function we use a [`ModuleMap`](https://github.com/amplication/amplication/blob/master/libs/util/code-gen-types/src/code-gen-types.ts).

Reviewer: ```suggestion
In the `after` function for .NET plugins, we use a [`FileMap`](https://github.com/amplication/amplication/blob/master/libs/util/code-gen-types/src/files/file-map.ts). For Node.js plugins' `after` function we use a [`ModuleMap`](https://github.com/amplication/amplication/blob/master/libs/util/code-gen-types/src/code-gen-types.ts#L149).
```
